src/main/java/io/cryostat/rules/PeriodicArchiver.java

```diff
@@ -102,86 +96,55 @@ public void run() {
             }
 
             performArchival();
-        } catch (InterruptedException | ExecutionException e) {
+        } catch (InterruptedException | ExecutionException | RecordingNotFoundException e) {
```

Comment:
I think this can be refactored - each of these exception types could be moved down into the general `catch (Exception)` block and added to the `if (hasCause(SomeException.class))` handling, and it would behave the same, right?

Reply:
Good idea, will do